### PR TITLE
Added Gemini Exchange and resolved conflicts

### PR DESCRIPTION
**Pull request by Rajat-S-Patel**

### CODE REVIEW

File: src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
+
+@Component
+public class ExcGemini implements ExcParent {
+
+    @Value("${api.exchange.gemini.baseurl}")
+    private String baseUrl;
+    private long getNonce(){
```
Comment (Mahendrasinh2401) [inline src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java]: can we create nonce method as util method?

File: src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        return new Date().getTime();
+    }
+    private byte[] getB64(String payload){
```
Comment (Mahendrasinh2401) [inline src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java]: got B64 can be reuse so can we create this method as util

File: src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/dto/request/CreateOrderRequest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.pirimid.cryptotrade.helper.exchange.gemini.dto.request;
+
```
Comment (Mahendrasinh2401) [inline src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/dto/request/CreateOrderRequest.java:2]: this DTO is used to send request to exchange. right?

File: src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/dto/response/CreateOrderResponse.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package com.pirimid.cryptotrade.helper.exchange.gemini.dto.response;
```
Comment (Mahendrasinh2401) [inline src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/dto/response/CreateOrderResponse.java:1]: this DTO is for getting response from exchange.right?

File: src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/dto/request/CreateOrderRequest.java (hunk 3 above)
Comment (Rajat-S-Patel) [reply src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/dto/request/CreateOrderRequest.java:2]: Yes it is for Gemini only

File: src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/dto/response/CreateOrderResponse.java (hunk 4 above)
Comment (Rajat-S-Patel) [reply src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/dto/response/CreateOrderResponse.java:1]: Yes

File: src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java (hunk 1 above)
Comment (Rajat-S-Patel) [reply src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java]: Yes

File: src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java (hunk 2 above)
Comment (Rajat-S-Patel) [reply src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java]: Yes

**Comment by Rajat-S-Patel** [comment]
Any other suggestions?

File: src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
```
Comment (HM1868) [inline src/main/java/com/pirimid/cryptotrade/helper/exchange/gemini/ExcGemini.java]: Can you remove this * ?

File: src/main/java/com/pirimid/cryptotrade/util/ExchangeUtil.java [hunk 6]
```diff
@@ -3,20 +3,24 @@
...
 import com.pirimid.cryptotrade.helper.exchange.coinbase.ExcCoinbase;
 import com.pirimid.cryptotrade.helper.exchange.ExcParent;
+import com.pirimid.cryptotrade.helper.exchange.gemini.ExcGemini;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class ExchangeUtil {
     @Autowired
     public ExcCoinbase exc_coinbase;
```
Comment (HM1868) [inline src/main/java/com/pirimid/cryptotrade/util/ExchangeUtil.java]: can you rename it using variable naming convention?

**Pull request merged**